File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/ui/AbstractMobileMainView.java
/**
 * This file is part of mycollab-mobile.
 *
 * mycollab-mobile is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-mobile is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-mobile.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.mobile.ui;

import com.esofthead.mycollab.vaadin.mvp.PageView;
import com.esofthead.mycollab.vaadin.mvp.ViewEvent;
import com.vaadin.ui.ComponentContainer;
import com.vaadin.ui.VerticalLayout;

import java.io.Serializable;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public abstract class AbstractMobileMainView extends VerticalLayout implements PageView, Serializable {
    private static final long serialVersionUID = 1L;

    @Override
    public void addViewListener(ViewListener listener) {
        addListener(ViewEvent.VIEW_IDENTIFIER(), ViewEvent.class, listener, ViewListener.viewInitMethod);
    }
}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/ui/AbstractMobileMainView.kt
package com.mycollab.mobile.ui

import com.mycollab.vaadin.event.ViewEvent
import com.mycollab.vaadin.mvp.PageView
import com.vaadin.ui.VerticalLayout

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
abstract class AbstractMobileMainView : VerticalLayout(), PageView {
    override fun <E> addViewListener(listener: PageView.ViewListener<E>) {
        addListener(ViewEvent.VIEW_IDENTIFIER, ViewEvent::class.java, listener, PageView.ViewListener.viewInitMethod)
    }
}